connection.go

```diff
@@ -739,7 +741,15 @@ func (s *Connection) shutdown(closeTimeout time.Duration) {
 	}
 
 	if err != nil {
```

Comment:
so, at this point we now the underly connection was close, there is no longer a TCP stream because s.conn.Close() and all streams were closed

Reply:
These waits for the shutdownChan to be drained, that can be done with the CloseWait() and Wait() methods, if this is not handled we just bail out. So effectively this is waiting for those two methods to be called.

The only way I can see this can not be handled is:
1. We call Close() directly
2. we receive a goaway frame

so the users need to invoke the Wait() method to handle this situation

Reply:
I'm just stating my understanding that matches the one in Jordan's description of the issue.

Reply:
thanks @aojea